File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int>mapp;
        for(int i=0; i<nums.size(); i++){
            int x=nums[i];
            if(mapp.find(target-x)!=mapp.end()){
                return {mapp[target-x], i};
            }
            mapp[x]=i;
        }
        return {};
    }
};

File: 1-two-sum/two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const mapp = new Map();

    for(let i=0; i<nums.length; i++){
        let num = nums[i];
        if(mapp.has(target-num)){
            return [mapp.get(target-num), i];
        }
        mapp.set(num,i);
    }
};